enforcer/src/main/java/org/wso2/micro/gateway/enforcer/config/ConfigHolder.java

```diff
@@ -37,38 +37,39 @@
 import java.security.NoSuchAlgorithmException;
 import java.security.cert.Certificate;
 import java.security.cert.CertificateException;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Properties;
 
 /**
  * Configuration holder class for Microgateway.
  */
-public class MGWConfiguration {
+public class ConfigHolder {
 
-    private static final Logger logger = LogManager.getLogger(MGWConfiguration.class);
+    private static final Logger logger = LogManager.getLogger(ConfigHolder.class);
 
-    private static MGWConfiguration mgwConfiguration;
-    private static Toml configToml;
-    private static Map<String, TokenIssuerDto> issuersMap;
-    private static EventHubConfigurationDto eventHubConfiguration;
-    private static KeyStore trustStore = null;
+    private static ConfigHolder configHolder;
+    private Toml configToml;
+    EnforcerConfig config;
+    private KeyStore trustStore = null;
 
-    private MGWConfiguration() throws MGWException {
+    private ConfigHolder() throws MGWException {
         try {
             init();
         } catch (KeyStoreException e) {
             throw new MGWException("Error while loading configuration from file", e);
         }
     }
 
-    public static MGWConfiguration getInstance() throws MGWException {
-        if (mgwConfiguration != null) {
-            return mgwConfiguration;
+    public static ConfigHolder getInstance() {
+        if (configHolder != null) {
+            return configHolder;
         }
-        mgwConfiguration = new MGWConfiguration();
-        return mgwConfiguration;
+        try {
+            configHolder = new ConfigHolder();
+        } catch (MGWException e) {
+            logger.error("Error while reading configuration", e);
```

Comment:
Shouldn't we throw here? Otherwise in an error scenario it'll cause and NPE when this object is accessed

Reply:
This is during server startup, if we throw from here that means we will have to kill the server if there is an issue in config reading, we need to decide whether we need to kill the server in case of an failure decide

Reply:
Yes thats correct. but if we didn't kill. That causes an NPE any where we read config after the bootup isn't it? so that again kills the server

Reply:
I think its better to handle the case in those places rather than killing the server. Killing the server for any config error is not good in my opinion. 
And also the places where we read this configs will not kill the server in case of an NPE since those are worker threads, Not the main thread